cmd/minikube/cmd/config/disable.go

```diff
@@ -38,11 +40,24 @@ var addonsDisableCmd = &cobra.Command{
 		if addon == "heapster" {
 			exit.Message(reason.AddonUnsupported, "The heapster addon is depreciated. please try to disable metrics-server instead")
 		}
-		err := addons.SetAndSave(ClusterFlagValue(), addon, "false")
+		cc, err := config.Load(ClusterFlagValue())
 		if err != nil {
-			exit.Error(reason.InternalAddonDisable, "disable failed", err)
+			exit.Error(reason.InternalAddonDisable, "loading profile", err)
+		}
+		validAddon, ok := assets.Addons[addon]
+		if ok {
```

Comment:
nit: With the change of the non-zero exit code below this `ok` can be inverted and reduce the nesting.

**Before:**
```
if ok {
	if validAddon.IsEnabled(cc) {
		err := addons.SetAndSave(ClusterFlagValue(), addon, "false")
		if err != nil {
			exit.Error(reason.InternalAddonDisable, "disable failed", err)
		}
	} else {
		out.Step(style.AddonDisable, `"The '{{.minikube_addon}}' addon is disabled`, out.V{"minikube_addon": addon})
	}

} else {
	out.Step(style.AddonDisable, `"The '{{.minikube_addon}}' seems not to be a valid minikube addon`, out.V{"minikube_addon": addon})
}
```

**After:**
```
if !ok {
	exit.Message(reason.InternalAddonDisable, `"'{{.minikube_addon}}' is not a valid minikube addon`, out.V{"minikube_addon": addon})
}
if validAddon.IsEnabled(cc) {
	err := addons.SetAndSave(ClusterFlagValue(), addon, "false")
	if err != nil {
		exit.Error(reason.InternalAddonDisable, "disable failed", err)
	}
}
out.Styled(style.AddonDisable, `"The '{{.minikube_addon}}' addon is disabled`, out.V{"minikube_addon": addon})
```